test/libsolidity/semanticTests/various/selfdestruct_post_cancun_create2.sol

```diff
@@ -0,0 +1,41 @@
+contract C {
+    constructor() payable {}
+
+    function terminate() external {
+        selfdestruct(payable(msg.sender));
+    }
+}
+
+contract D {
+    C public c;
+
+    constructor() payable {
+        c = new C{value: 1 ether, salt: hex"1234"}();
+    }
+
+    function destroy() external {
+        assert(address(this).balance == 0);
+        assert(address(c).balance == 1 ether);
+        assert(exists());
+        c.terminate();
+    }
+
+    function testSelfDestruct() external view {
+        assert(exists()); // NOTE: the contract should still exists
+        assert(address(this).balance == 1 ether);
+        assert(address(c).balance == 0);
+    }
```

Comment:
Why not make these expectations like in other tests? Or, at worst, just return the three values from the function instead of asserting and then reuse it for `destroy()` too?

Reply:
Because using salted create actually does not yield the same address in our test framework when using the test expectation that require an address, like: `// balance: <address> -> 0`. This happens because the address calculation depends on the exact init code and that will change between legacy and via-IR codegen and via optimized vs non-optimized.